rdr_service/dao/genomic_datagen_dao.py

```diff
@@ -195,11 +197,28 @@ def get_template_by_name(self, project_name, template_name):
                 GenomicDataGenManifestSchema.ignore_flag == 0
             ).order_by(GenomicDataGenManifestSchema.field_index).all()
 
-    def execute_manifest_query(self, columns, sample_ids):
+    def execute_manifest_query(self, columns, sample_ids, cvl_id=None):
+
         with self.session() as session:
-            return session.query(
+            query = session.query(
                 *columns
             ).join(
                 ParticipantSummary,
                 ParticipantSummary.participantId == GenomicSetMember.participantId
-            ).filter(GenomicSetMember.sampleId.in_(sample_ids)).all()
+            ).outerjoin(
+                GenomicGCValidationMetrics,
+                GenomicGCValidationMetrics.genomicSetMemberId == GenomicSetMember.id
+            ).outerjoin(
+                GenomicCVLSecondSample,
+                GenomicCVLSecondSample.genomic_set_member_id == GenomicSetMember.id
+            ).outerjoin(
+                GenomicCVLAnalysis,
+                GenomicCVLAnalysis.genomic_set_member_id == GenomicSetMember.id
+            ).filter(GenomicSetMember.sampleId.in_(sample_ids))
+
+            if cvl_id:
+                query_dao = GenomicQueriesDao()
+                gc_site_id = query_dao.transform_cvl_site_id(cvl_id)
+                query = query.filter(GenomicSetMember.gcSiteId == gc_site_id)
```

Comment:
Thanks for doing this quickly.

We can convert the `transform_cvl_site_id` to a class method and then interact with it as `GenomicQueriesDao.transform_cvl_site_id` instead of having to instantiate the entire `GenomicQueriesDao` class.

Reply:
True that, I'll make that change in a separate PR so that we're not holding up testing.